socket rooms added

devongillis opened the pull request.

devongillis committed:
socket rooms added

wpasadilla commented on poker-pals/src/components/pokerComponents/RaiseToggle.vue:
```diff
@@ -3,8 +3,8 @@
-        <div class="raiseValue" id="raiseValue"> {{ bigBlind }}</div>
-        <input class="raiseScroll" id="slider" type="range" v-bind:min="bigBlind" v-bind:max="max" v-bind:value="bigBlind"> <!--TODO: set these using props-->
+        <div class="raiseValue">{{ raiseValue }}</div>
+        <input class="raiseScroll" id="slider" type="range" v-bind:min="bigBlind" v-bind:max="bigBlind*20" v-bind:value="raiseValue"> <!--TODO: set these using props-->
```
Not sure if this another one of those magic number cases 

bigBlind*20 

Note: it was not brought up during the last pull request.

devongillis replied:
nice catch

devongillis committed:
magic number fix

darbrius commented on poker-pals/src/components/Poker.vue:
```diff
@@ -217,8 +208,8 @@ export default {
-            let b = document.getElementById("buttonCheckFold");
-            b.style.backgroundColor = '#ffffff'; // color theme for button not selected
+            //let b = document.getElementById("buttonCheckFold");
+            //b.style.backgroundColor = '#ffffff'; // color theme for button not selected
```
Don't forget to remove commented code

wpasadilla commented on server/app.js:
```diff
@@ -62,13 +62,19 @@ let profiles = [
+    {userID: 5000, name: 'Stan693', chips: 100000, icon: 'player_icon_1'},
     {userID: 6000, name: 'Mack273', chips: 20, icon: 'player_icon_1'},
 ];
 // the entire list of tables, might need a better way to store them?
```
I had it set up like so using a JS object. Rooms are all given a unique ID and then stored into the rooms object using their ID as a map.  

let rooms = {};
for (let x = 1; x <= 10; x++) {
        //Create x number of rooms 
        const room = {
            id: uuidv4(),
            name: "Table " + x,
            buyin: 20000,
            blinds: "1000 / 2000",
            sockets: []
        }
        rooms[room.id] = room;
    }

wpasadilla replied:
Maybe this is a more scale-able approach? 

Need opinions.

devongillis replied:
we definitely need to address this tomorrow.

darbrius commented on server/app.js:
```diff
@@ -78,7 +84,7 @@ function login(socket, msg){ // user sends a log in request and the server check
     }
     else{// send the current table state
-        socket.emit('tableState', JSON.stringify(tables[0].getTableState()));
+        //socket.emit('tableState', JSON.stringify(tables[0].getTableState()));
```
More commented code to remove

wpasadilla commented on server/app.js:
```diff
@@ -78,7 +84,7 @@ function login(socket, msg){ // user sends a log in request and the server check
     }
     else{// send the current table state
-        socket.emit('tableState', JSON.stringify(tables[0].getTableState()));
+        //socket.emit('tableState', JSON.stringify(tables[0].getTableState()));
```
Is this still being implemented? 
Or just commented out for testing purposes?

devongillis replied:
no, i realized that the user would not be sent the table state on login, they first would see the list of tables. one thing we really need to get done is linking up all the pages together so i can start using real authentication and real profiles. we should address this in tomorrows meeting

devongillis committed:
fixes

aidan-lee commented on poker-pals/src/components/Poker.vue:
```diff
@@ -215,10 +205,7 @@ export default {
             });
-            // TODO: later use v-bind and not getElementById
             this.checkFold = false;
-            let b = document.getElementById("buttonCheckFold");
-            b.style.backgroundColor = '#ffffff'; // color theme for button not selected
```
Could b have a more descriptive name?

devongillis replied:
that code was removed

aidan-lee replied:
Weird.  Totally saw it as an addition.  My mistake.

devongillis committed:
uses real profiles, poker is connected to tables

poker.vue is connected to tables.vue but tables.vue still needs to connect to poker.vue (and send the table id and user id)

devongillis committed:
added rooms

devongillis committed:
fixed socket room disconnect

devongillis committed:
Merge branch 'master' into socket-rooms

devongillis committed:
solved merge conflict
